src/torrent.cpp

```diff
@@ -5784,17 +5762,11 @@ namespace {
 	{
 		INVARIANT_CHECK;
 
-		if (!valid_metadata())
-		{
-			files->resize(m_file_priority.size());
-			std::copy(m_file_priority.begin(), m_file_priority.end(), files->begin());
-			return;
-		}
+		files->assign(m_file_priority.begin(), m_file_priority.end());
```

Comment:
this new code is not quite equivalent. previously the size of `files` was preserved. Now it isn't. Was that deliberate?